add missing types

anonrig opened the pull request:
Running the following command on main branch creates these extra types which makes it extremely hard for new PRs to pass the CI

```
bazel build //types && rm -rf types/generated-snapshot && cp -r bazel-bin/types/definitions types/generated-snapshot
```

npaun reviewed (approved):
Thank you for your service 🫡

anonrig commented:
cc @penalosa any idea why the CI is failing?

penalosa commented:
@anonrig I can't replicate this—running the provided command on this branch for me _removes_ the types added (which is probably why CI is failing

anonrig commented:
> bazel build //types && rm -rf types/generated-snapshot && cp -r bazel-bin/types/definitions types/generated-snapshot

I can still replicate this on main branch

```
workerd on  main via ⬢ v23.8.0
❯ pnpm i
Lockfile is up to date, resolution step is skipped
Already up to date

   ╭──────────────────────────────────────────────────────────────────╮
   │                                                                  │
   │                Update available! 9.7.1 → 10.5.2.                 │
   │   Changelog: https://github.com/pnpm/pnpm/releases/tag/v10.5.2   │
   │         Run a script from: https://pnpm.io/installation          │
   │                                                                  │
   │         Follow @pnpmjs for updates: https://x.com/pnpmjs         │
   │                                                                  │
   ╰──────────────────────────────────────────────────────────────────╯

Done in 339ms

workerd on  main via ⬢ v23.8.0
❯ bazel build //types && rm -rf types/generated-snapshot && cp -r bazel-bin/types/definitions types/generated-snapshot
INFO: Analyzed target //types:types (475 packages loaded, 26704 targets configured).
INFO: Found 1 target...
Target //types:types up-to-date:
  bazel-bin/types/definitions
INFO: Elapsed time: 80.946s, Critical Path: 76.65s
INFO: 154 processes: 5 internal, 140 linux-sandbox, 9 local.
INFO: Build completed successfully, 154 total actions

workerd on  main via ⬢ v23.8.0 took 1m21s
❯ git status
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   types/generated-snapshot/2021-11-03/index.d.ts
        modified:   types/generated-snapshot/2021-11-03/index.ts
        modified:   types/generated-snapshot/2022-01-31/index.d.ts
        modified:   types/generated-snapshot/2022-01-31/index.ts
        modified:   types/generated-snapshot/2022-03-21/index.d.ts
        modified:   types/generated-snapshot/2022-03-21/index.ts
        modified:   types/generated-snapshot/2022-08-04/index.d.ts
        modified:   types/generated-snapshot/2022-08-04/index.ts
        modified:   types/generated-snapshot/2022-10-31/index.d.ts
        modified:   types/generated-snapshot/2022-10-31/index.ts
        modified:   types/generated-snapshot/2022-11-30/index.d.ts
        modified:   types/generated-snapshot/2022-11-30/index.ts
        modified:   types/generated-snapshot/2023-03-01/index.d.ts
        modified:   types/generated-snapshot/2023-03-01/index.ts
        modified:   types/generated-snapshot/2023-07-01/index.d.ts
        modified:   types/generated-snapshot/2023-07-01/index.ts
        modified:   types/generated-snapshot/experimental/index.d.ts
        modified:   types/generated-snapshot/experimental/index.ts
        modified:   types/generated-snapshot/oldest/index.d.ts
        modified:   types/generated-snapshot/oldest/index.ts

no changes added to commit (use "git add" and/or "git commit -a")
```

anonrig commented:
Apparently this is a local cache bug.
